pyvrp/cpp/search/PerturbationOperator.h

```diff
@@ -3,12 +3,28 @@
 
 #include "CostEvaluator.h"
 #include "DynamicBitset.h"
+#include "LocalSearchOperator.h"
 #include "Route.h"
 
 #include <vector>
 
 namespace pyvrp::search
 {
+/**
+ * Context structure that holds all parameters needed by perturbation operators.
+ */
+struct PerturbationContext
+{
+    std::vector<search::Route::Node> &nodes;
+    std::vector<search::Route> &routes;
+    CostEvaluator const &costEvaluator;
+    std::vector<std::vector<size_t>> const &neighbours;
+    std::vector<size_t> const &orderNodes;
+    std::vector<std::pair<size_t, size_t>> const &orderVehTypes;
+    DynamicBitset &promising;
+    size_t const numPerturbations;
+};
```

Comment:
I think this is OK to have for now as we still figure out how to develop ILS, but I'd like to get rid of this again at some point.